### PR TITLE
Update DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by norbiorb**
This PR includes the documentation of https://github.com/adobe/aem-sample-we-retail-journal/issues/13#issuecomment-432779863 into [DEVELOPMENT.md](https://github.com/adobe/aem-sample-we-retail-journal/blob/development/react-app/DEVELOPMENT.md) for react-app and angular-app

It updates also the versions for maven surefire and failsafe plugin to include the fix for [SUREFIRE-1588](https://issues.apache.org/jira/browse/SUREFIRE-1588)

### CODE REVIEW

File: react-app/DEVELOPMENT.md [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
 ```
-This will open the local dev server on port 9000. You can navigate then to http://localhost:9000/content/we-retail-journal/react/home.html
-
+This will open the local dev server on port 9000. You can navigate then to `http://localhost:9000/content/we-retail-journal/react/home.html`
+where `API_HOST` points to your AEM instance.
 ### GOTCHA's - you should read this
 
 #### Your requests to AEM are failing
-Most probably this is due to CORS rejection, so you might want to either config AEM to accept CORS or bypass it from the browser if you are doing local development.  
+Most probably this is due to CORS rejection, so you might want to either config AEM to accept CORS or bypass it from the browser if you are doing local development.
```
Comment (pfauchere) [inline react-app/DEVELOPMENT.md]: `config` > `configure`

File: react-app/DEVELOPMENT.md [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
-This will open the local dev server on port 9000. You can navigate then to http://localhost:9000/content/we-retail-journal/react/home.html
-
+This will open the local dev server on port 9000. You can navigate then to `http://localhost:9000/content/we-retail-journal/react/home.html`
+where `API_HOST` points to your AEM instance.
 ### GOTCHA's - you should read this
 
 #### Your requests to AEM are failing
-Most probably this is due to CORS rejection, so you might want to either config AEM to accept CORS or bypass it from the browser if you are doing local development.  
+Most probably this is due to CORS rejection, so you might want to either config AEM to accept CORS or bypass it from the browser if you are doing local development. 
 **Also** make sure you login in AEM instance in the same browser.
```
Comment (pfauchere) [inline react-app/DEVELOPMENT.md]: I wonder if that is necessary

File: react-app/DEVELOPMENT.md (hunk 1 above)
Comment (lydiapuric) [reply react-app/DEVELOPMENT.md]: angular-app/DEVELOPMENT.md needs this updated too

File: react-app/DEVELOPMENT.md [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
 
+#### Authorize your requests
+* Instantiate a CustomModelClient such as [react-app/src/server/CustomModelClient.js](https://github.com/adobe/aem-sample-we-retail-journal/blob/master/react-app/src/server/CustomModelClient.js) 
+when initializing the [ModelManager in index.js](https://github.com/adobe/aem-sample-we-retail-journal/blob/master/react-app/src/index.js#L42)
+* If necessary adapt Authorization header in [react-app/src/server/CustomModelClient.js](https://github.com/adobe/aem-sample-we-retail-journal/blob/master/react-app/src/server/CustomModelClient.js#L21)
```
Comment (pfauchere) [inline react-app/DEVELOPMENT.md]: adapt *the* Authorization header

File: react-app/DEVELOPMENT.md [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
+Most probably this is due to CORS rejection, so you might want to either config AEM to accept CORS or bypass it from the browser if you are doing local development. 
 **Also** make sure you login in AEM instance in the same browser.
 
+#### Authorize your requests
+* Instantiate a CustomModelClient such as [react-app/src/server/CustomModelClient.js](https://github.com/adobe/aem-sample-we-retail-journal/blob/master/react-app/src/server/CustomModelClient.js) 
+when initializing the [ModelManager in index.js](https://github.com/adobe/aem-sample-we-retail-journal/blob/master/react-app/src/index.js#L42)
+* If necessary adapt Authorization header in [react-app/src/server/CustomModelClient.js](https://github.com/adobe/aem-sample-we-retail-journal/blob/master/react-app/src/server/CustomModelClient.js#L21) 
+
+#### Update the CORS configuration of the AEM instance
+* Console Path: http://localhost:4502/system/console/configMgr
```
Comment (pfauchere) [inline react-app/DEVELOPMENT.md]: I would switch the formatting from bullet points to numbers

Comment (pfauchere) [reply react-app/DEVELOPMENT.md]: 1. Navigate to the Configuration Manager on the AEM instance http://localhost:4502/system/console/configMgr

**Comment by pfauchere** [comment]
We have an issue with Surefire

`Could not find or load main class org.apache.maven.surefire.booter.ForkedBooter`

https://circleci.com/gh/adobe/aem-sample-we-retail-journal/26

Can you please investigate [0]?

[0] https://stackoverflow.com/a/53016532

File: react-app/DEVELOPMENT.md (hunk 2 above)
Comment (norbiorb) [reply react-app/DEVELOPMENT.md]: No, not necessary at least when having configured CORS

File: react-app/DEVELOPMENT.md (hunk 1 above)
Comment (norbiorb) [reply react-app/DEVELOPMENT.md]: Yes that is true, at least with these common setup steps

**Comment by norbiorb** [comment]
Setting useSystemClassLoader to false as mentioned in http://maven.apache.org/surefire/maven-surefire-plugin/examples/class-loading.html could help.

**Review by pfauchere — APPROVED** [review]

**Pull request merged**